From 3d19fce32f675e9596fe3da8ade09399be11f9fa Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Thu, 20 Dec 2012 14:25:53 +0800
Subject: [PATCH 02/25] ENGR00236896 do not init directoutput stream when
 audio policy init

do not init directoutput strean when audio policy init

Change-Id: I18f7a0a73e965ef4346c8f7d19eefb0790ef6443
Signed-off-by: b02247 <b02247@freescale.com>
---
 audio/AudioPolicyManagerBase.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index abfd143..53d0038 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -1587,8 +1587,7 @@ AudioPolicyManagerBase::AudioPolicyManagerBase(AudioPolicyClientInterface *clien
         {
             const IOProfile *outProfile = mHwModules[i]->mOutputProfiles[j];
 
-            if ((outProfile->mSupportedDevices & mAttachedOutputDevices) &&
-                    ((outProfile->mFlags & AUDIO_OUTPUT_FLAG_DIRECT) == 0)) {
+            if (outProfile->mSupportedDevices & mAttachedOutputDevices && !(outProfile->mFlags & AUDIO_OUTPUT_FLAG_DIRECT)) {
                 AudioOutputDescriptor *outputDesc = new AudioOutputDescriptor(outProfile);
                 outputDesc->mDevice = (audio_devices_t)(mDefaultOutputDevice &
                                                             outProfile->mSupportedDevices);
-- 
1.8.0

